var/spack/repos/builtin/packages/ncl/package.py

```diff
@@ -194,6 +239,7 @@ def prepare_site_config(self):
             f.writelines(
                 [
                     "#define HdfDefines\n",
+                    bytelines,
```

Comment:
Maybe
```suggestion
                    "#define StdDefines -DByteSwapped\n#define ByteSwapped\n" if "+byteswapped" in self.spec else "",
```
Interesting how black formats this :)

Reply:
Sorry, I missed the email notification for the original review.

The hdf-eos5 stuff looks OK to me.  Note however, that I am really not the best choice as a maintainer as I do not use the package, I just created the spack recipe in order to build the package for one of my users.  But I also put the chmod stuff in for hdf-eos2 (same sort of situation), and will likely steal your h5cc fix for hdf-eos2 as well (still giving me issues in a rebuild).

But in short, the hdf-eos5 PR looks good to me.

Reply:
Thanks @skosukhin and @payerle for the quick inspections. I'll plan to make the above change on the "bytelines" logic.

I'm in a similar boat with respect to hdf-eos packages - supporting them for users. They clearly don't get used in Spack very often given the permissions issues introduced in the past 1-2 years, though I suspect they will get a bit more attention as dependencies with us and spack-stack using them.

Reply:
@skosukhin - I made all of the suggested changes. I did keep the use of `spec.satisfies("+byteswapped")` instead of using `in`. I'll admit that I find the documented distinction a bit confusing, but from the discussion in the Spack slack yesterday I think I understand that using `in` checks the condition against dependencies too, and really what I care about here is whether NCL itself is configured in "byteswapped" mode, not any of its dependencies should they have a similar variant.

Let me know if my understanding is confused still!